src/services/server/station.service.ts

```
@@ -0,0 +1,275 @@
+      for (let i = 0; i < line.stations.length; i++) {
+        const station = line.stations[i];
+
+        if (!this.stations.has(station.id)) {
+          this.stations.set(station.id, station);
+          this.stationsArray.push(station);
+          this.spatialIndex.add(station);
+          this.stationLines.set(station.id, []);
+        }
+
+        // Associate line with station
+        const lines = this.stationLines.get(station.id)!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)